apis/config/v1beta1/configuration_types.go

```diff
@@ -588,3 +592,29 @@ type WorkloadRetentionPolicy struct {
 	// +optional
 	AfterDeactivatedByKueue *metav1.Duration `json:"afterDeactivatedByKueue,omitempty"`
 }
+
+type FailureRecoveryPolicy struct {
+	// Rules specifies the rules to be enabled for failure recovery.
+	Rules []FailureRecoveryRule `json:"rules"`
+}
+
+type FailureRecoveryRule struct {
+	// Exactly one of the fields below must be specified.
+
+	// TerminatePod enables and contains configuration for the `TerminatePod` strategy.
+	// This strategy recovers stuck pods by forcefully terminating them after a configured
+	// grace period elapses.
+	// +optional
+	TerminatePod *TerminatePodConfig `json:"terminatePod,omitempty"`
+}
+
+type TerminatePodConfig struct {
+	// PodLabelSelector specifies the scope of resources covered by `TerminatePod` failure recovery -
+	// resources not matching the selector are ignored by the controller.
+	PodLabelSelector *metav1.LabelSelector `json:"podLabelSelector,omitempty"`
+
+	// ForcefulTerminationGracePeriod is the duration between when the pod's `deletionGracePeriodSeconds`
+	// elapses and when the pod should be forcefully deleted.
+	// Represented using metav1.Duration (e.g. "10m", "1h30m").
+	ForcefulTerminationGracePeriod *metav1.Duration `json:"forcefulTerminationGracePeriod"`
```

Comment:
Quoting your KEP:

_The proposal is to **not** set a default value for the forceful termination grace period. This puts configuration over convention, as it will make the user consciously think about the grace period the need for their specific use case, without risking a faulty default._

This sounds like there should be some validation here against this being `nil`?
(maybe also against 0 seconds?)

BTW are there reasons for this to be a pointer? (non-pointer `Duration`s do exist in `_types.go`)